tidb: update tidb_index_serial_scan_concurrency docs (#21938)
pingcap/docs: system-variables.md

### tidb_analyze_distsql_scan_concurrency <span class="version-mark">New in v7.6.0</span>

> **Note:**
>
> Before v7.6.0, it was controlled by the variable `tidb_distsql_scan_concurrency`.

> > So please consider changing the value of `tidb_distsql_scan_concurrency` if you want to adjust the concurrency of scanning TiKV regions in versions earlier than v7.6.0.

Reviewer: After this PR, this variable covers both regular region scans and special-index scans. For versions earlier than v7.6.0, those two paths were controlled by different legacy variables, so pointing readers only to `tidb_distsql_scan_concurrency` loses the index-stats case.
```suggestion
> In versions earlier than v7.6.0, regular `ANALYZE` region scans are controlled by `tidb_distsql_scan_concurrency`, and index statistics scans are controlled by `tidb_index_serial_scan_concurrency`. Therefore, to adjust the concurrency of scanning TiKV regions for these versions, consider changing the value of `tidb_distsql_scan_concurrency`.
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Integer
- Default value: `4`
- Range: `[0, 4294967295]`. In versions earlier than v8.2.0, the minimum value is `1`. When you set it to `0`, it adaptively adjusts the concurrency based on the cluster size.
- Used to control the concurrency of scanning TiKV regions.

> - Used to control the concurrency of scanning regions for special indexes (indexes generated from virtual columns).

Reviewer: These two fragment bullets would read more naturally if introduced by a full sentence, which also keeps the multi-behavior variables in this PR consistent.
```suggestion
- This variable controls the following aspects of `ANALYZE` concurrency:
    - The concurrency of scanning TiKV regions.
    - The concurrency of scanning regions for special indexes, such as indexes on generated virtual columns.
```

>     - The concurrency of scanning regions for special indexes, such as indexes on generated virtual columns.

Reviewer: ```suggestion
    - The concurrency of scanning TiKV Regions.
    - The concurrency of scanning Regions for special indexes, such as indexes on generated virtual columns.
```

- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Default value: `2`. The default value is `1` for v7.4.0 and earlier versions.
- Range: `[1, 128]`. Before v8.4.0, the value range is `[1, 18446744073709551615]`.

> - For manual/auto analyze: used to control the concurrency for saving analyze results (writing TopN and histograms to system tables).

Reviewer: ```suggestion
- For manual and auto analyze, this variable controls the concurrency for saving analyze results, including writing TopN and histograms to system tables.
```

> - For manual and auto analyze, this variable controls the concurrency for saving analyze results, including writing TopN and histograms to system tables.

Reviewer: ```suggestion
- For manual and auto `ANALYZE`, this variable controls the concurrency for saving `ANALYZE` results, including writing TopN and histograms to system tables.
```

### tidb_analyze_distsql_scan_concurrency <span class="version-mark">New in v7.6.0</span>

> **Note:**
>

> > In versions earlier than v7.6.0, regular `ANALYZE` region scans are controlled by `tidb_distsql_scan_concurrency`, and index statistics scans are controlled by `tidb_index_serial_scan_concurrency`. Therefore, to adjust the concurrency of scanning TiKV regions for these versions, consider changing the value of `tidb_distsql_scan_concurrency`.

Reviewer: ```suggestion
> In versions earlier than v7.6.0, regular `ANALYZE` region scans are controlled by `tidb_distsql_scan_concurrency`, and index statistics scans are controlled by `tidb_index_serial_scan_concurrency`. Therefore, to adjust the concurrency of scanning TiKV Regions for these versions, consider changing the value of `tidb_distsql_scan_concurrency`.
```

- Type: Integer
- Default value: `1`
- Range: `[1, 256]`

> - For auto analyze: used to control the concurrency of table or partition analysis — for example, the number of partitions or table tasks that can be processed simultaneously.

Reviewer: ```suggestion
- For auto analyze, this variable controls the concurrency of table or partition analysis, for example, the number of partitions or table tasks that can be processed simultaneously.
```

> - For auto analyze, this variable controls the concurrency of table or partition analysis, for example, the number of partitions or table tasks that can be processed simultaneously.

Reviewer: ```suggestion
- This variable controls the concurrency for building statistics during auto `ANALYZE`, such as the number of table or partition analysis tasks that can be processed simultaneously
```

- Default value: `2`. The default value is `4` for v7.4.0 and earlier versions.
- Range: `[1, 256]`
- Unit: Threads

> - For manual analyze: used to control the concurrency of table or partition analysis — for example, the number of partitions or table tasks that can be processed simultaneously.

Reviewer: ```suggestion
- For manual analyze, this variable controls the concurrency of table or partition analysis, for example, the number of partitions or table tasks that can be processed simultaneously.
```

> - For manual analyze, this variable controls the concurrency of table or partition analysis, for example, the number of partitions or table tasks that can be processed simultaneously.

Reviewer: ```suggestion
- This variable controls the concurrency for building statistics during manual `ANALYZE`, such as the number of table or partition analysis tasks that can be processed simultaneously.
```

- Unit: Threads
- Default value: `2`
- Range: `[1, 256]`
- Used to control the concurrency level for merging samples collected from different regions.

> - Used to control the concurrency for special indexes (such as indexes on generated virtual columns), for example, the number of indexes which can concurrently collect statistics.

Reviewer: For the multi-behavior variables in this PR, could we introduce the list with a complete sentence? That would make this section parallel with the `statistics.md` wording and easier to scan.
```suggestion
- This variable controls the following aspects of `ANALYZE` concurrency:
    - The concurrency for merging samples collected from different regions.
    - The concurrency for special indexes (such as indexes on generated virtual columns), for example, the number of indexes that TiDB can concurrently collect statistics for.
```

- Unit: Threads
- Default value: `2`
- Range: `[1, 256]`
- This variable controls the following aspects of `ANALYZE` concurrency:

>     - The concurrency for merging samples collected from different regions.

Reviewer: ```suggestion
    - The concurrency for merging samples collected from different Regions.
```

>     - The concurrency for special indexes (such as indexes on generated virtual columns), for example, the number of indexes that TiDB can concurrently collect statistics for.

Reviewer: ```suggestion
    - The concurrency for collecting statistics on special indexes (such as indexes on generated virtual columns), for example, the number of indexes that TiDB can concurrently collect statistics for.
```

### tidb_index_serial_scan_concurrency

> **Warning:**
>

> > This variable is deprecated and cannot be modified. Sequential index scans now follow [`tidb_executor_concurrency`](#tidb_executor_concurrency-new-in-v50), and [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) uses [`tidb_analyze_distsql_scan_concurrency`](#tidb_analyze_distsql_scan_concurrency-new-in-v760) to control index read concurrency.

Reviewer: The TiDB change this PR references still allows `SET @@tidb_index_serial_scan_concurrency = ...`; it just emits a deprecation warning and no longer affects execution behavior. Saying it "cannot be modified" is stronger than the product behavior and can mislead users who test it interactively.
```suggestion
> **Warning:**
>
> This variable is deprecated and no longer controls execution behavior. Sequential index scans now follow [`tidb_executor_concurrency`](#tidb_executor_concurrency-new-in-v50), and [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) uses [`tidb_analyze_distsql_scan_concurrency`](#tidb_analyze_distsql_scan_concurrency-new-in-v760) to control index read concurrency.
```

### tidb_index_serial_scan_concurrency

> **Warning:**
>

> > This variable is deprecated and no longer controls execution behavior. Sequential index scans now follow [`tidb_executor_concurrency`](#tidb_executor_concurrency-new-in-v50), and [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) uses [`tidb_analyze_distsql_scan_concurrency`](#tidb_analyze_distsql_scan_concurrency-new-in-v760) to control index read concurrency.

Reviewer: ```suggestion
> This variable is deprecated and no longer controls execution behavior. The concurrency of sequential index scans is now controlled by [`tidb_executor_concurrency`](#tidb_executor_concurrency-new-in-v50), and [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) uses [`tidb_analyze_distsql_scan_concurrency`](#tidb_analyze_distsql_scan_concurrency-new-in-v760) to control the concurrency of index statistics scans.
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
- Type: Integer
- Default value: `1`
- Range: `[1, 256]`
- Unit: Threads

> - This variable remains only for backward compatibility. Adjust [`tidb_executor_concurrency`](#tidb_executor_concurrency-new-in-v50) to affect sequential index scans, or [`tidb_analyze_distsql_scan_concurrency`](#tidb_analyze_distsql_scan_concurrency-new-in-v760) to tune index statistics collection.

Reviewer: ```suggestion
- This variable remains only for backward compatibility. Use [`tidb_executor_concurrency`](#tidb_executor_concurrency-new-in-v50) to control the concurrency of sequential index scans, or [`tidb_analyze_distsql_scan_concurrency`](#tidb_analyze_distsql_scan_concurrency-new-in-v760) to control the concurrency of index statistics scans.
```

- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Default value: `1`

> - Used to control the concurrency for merging TopN results of partitioned tables.

Reviewer: ```suggestion
- This variable controls the concurrency for merging TopN results of partitioned tables.
```
